### ISSUE TITLE
如何训练yolov_x_online的模型

### ISSUE THREAD

**Issue by Eudian** [YuHengsss/YOLOV#92]
试了yolov_l_online, yolov_s_online的训练都跑不起来。

**Comment by YuHengsss** [comment]
These files are used for inferencing, please use config like yolov_l or yolov_s for training!

**Comment by Eudian** [comment]
So train with yolov_x.py, and inferencing with yolov_x_online.py, it goes in this way?

**Comment by YuHengsss** [comment]
yep!


---- 回复的原邮件 ----
发件人  ***@***.***> 日期  2024年08月06日 14:02 收件人  ***@***.***> 抄送至  ***@***.***>***@***.***> 主题  Re: [YuHengsss/YOLOV] 如何训练yolov_x_online的模型 (Issue #92)  

So train with yolov_x.py, and inferencing with yolov_x_online.py, it goes in this way?

**Comment by Johnny-Lin** [comment]
class MSA_yolov_online(nn.Module):

    def __init__(self, dim,out_dim, num_heads=4, qkv_bias=False, attn_drop=0.,scale=25):
        super().__init__()
        self.msa = Attention_msa_online(dim,num_heads,qkv_bias,attn_drop,scale=scale)
        self.linear1 = nn.Linear(2 * dim,2 * dim)
        self.linear2 =  nn.Linear(4 * dim,out_dim)

    def ave_pooling_over_ref(self,features,sort_results):
        key_feature = features[0]
        support_feature = features[0]
        if not self.training:
            sort_results = sort_results.to(features.dtype)
        soft_sim_feature = (sort_results@support_feature)#.transpose(1, 2)#torch.sum(softmax_value * most_sim_feature, dim=1)
        cls_feature = torch.cat([soft_sim_feature,key_feature],dim=-1)

        return cls_feature
那这里推理用到的参数是怎么来的 ？
